crates/core/src/db/relational_db.rs

```diff
@@ -724,48 +792,154 @@ impl RelationalDB {
     }
 }
 
-fn make_default_ostorage(in_memory: bool, path: impl AsRef<Path>) -> Result<Box<dyn ObjectDB + Send>, DBError> {
-    Ok(if in_memory {
-        Box::<MemoryObjectDB>::default()
-    } else {
-        Box::new(HashMapObjectDB::open(path)?)
-    })
-}
-
-pub fn open_db(path: impl AsRef<Path>, in_memory: bool, fsync: bool) -> Result<RelationalDB, DBError> {
-    let path = path.as_ref();
-    let mlog = if in_memory {
-        None
-    } else {
-        Some(Arc::new(Mutex::new(
-            MessageLog::open(path.join("mlog")).map_err(|e| DBError::Other(e.into()))?,
-        )))
-    };
-    let odb = Arc::new(Mutex::new(make_default_ostorage(in_memory, path.join("odb"))?));
-    let stdb = RelationalDB::open(path, mlog, odb, Address::zero(), fsync)?;
-
-    Ok(stdb)
-}
-
-pub fn open_log(path: impl AsRef<Path>) -> Result<Arc<Mutex<MessageLog>>, DBError> {
-    let path = path.as_ref().to_path_buf();
-    Ok(Arc::new(Mutex::new(
-        MessageLog::open(path.join("mlog")).map_err(|e| DBError::Other(e.into()))?,
-    )))
-}
-
 #[cfg(test)]
 pub(crate) mod tests_utils {
+    use std::fs::create_dir_all;
+    use std::ops::Deref;
+
     use super::*;
     use tempfile::TempDir;
 
-    // Utility for creating a database on a TempDir
-    pub(crate) fn make_test_db() -> Result<(RelationalDB, TempDir), DBError> {
-        let tmp_dir = TempDir::with_prefix("stdb_test")?;
-        let in_memory = false;
-        let fsync = false;
-        let stdb = open_db(&tmp_dir, in_memory, fsync)?.with_row_count(Arc::new(|_, _| i64::MAX));
-        Ok((stdb, tmp_dir))
+    /// A [`RelationalDB`] in a temporary directory.
+    ///
+    /// When dropped, any resources including the temporary directory will be
+    /// deleted.
+    ///
+    /// To keep the temporary directory, use [`Self::reopen`] or [`Self::into_parts`].
+    ///
+    /// [`TestDB`] is deref-coercible into [`RelationalDB`], which is dubious
+    /// but convenient.
+    pub struct TestDB {
```

Comment:
TODO: This should go into the `testing` crate (probably), so bugs around dropping the runtime are made in a single place.